### PR TITLE
fix(FocusTrapZone): aria-hidden on FocusTrapZone's parent is not removed when there's multiple FTZ

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
## tl;dr
FocusTrapZone adds aria-hidden to only direct children of body. But when it's time to remove aria-hidden, it only tries to remove aria-hidden on FocusTrapZone element itself, even though it is not always a direct children of body. In this case the aria-hidden on the parent element is never removed.
This PR checks for parent element of FocusTrapZone and remove aria-hidden

## A long and painful story of how I came to this problem 🥲:
Try: https://codesandbox.io/s/confident-rosalind-dhi2gy?file=/example.tsx

1. click on the button to open popup, notice a dialog is opened at the same time
2. dismiss dialog by clicking on 'cancel' button
3. hit 'tab' key and focus should go inside the first button 'grid-item-0' in Popup content.
4. try to navigate by right arrow key. The expected behavior is that focus move to 'grid-item-1', but the actual behavior is focus does not move.

Now there's one thing that makes arrow navigation starts to work:
After dismissing Dialog, open browser devtools, find the outer most wrapper of the popup content. It should be an element with className 'ui-provider', and attribute `aria-hidden="true" data-is-hidden-from-acc-tree="true"`. Removing `aria-hidden="true"` will make arrow navigation starts to work:

https://user-images.githubusercontent.com/28751745/185427650-c29b545c-045b-4758-b90c-0c7431ad536e.mov

This happens because the arrow navigation here is done by tabster mover. And it won't work when there's a 'aria-hidden' element in the parent of the mover element.

To sum it up, the `aria-hidden` on Popup wrapper div should go away when dialog is dismissed. But it wasn't.

Here's a more abstract example with just FocusTrapZone: https://codesandbox.io/s/distracted-panna-uwihrz?file=/example.tsx
Note that when click on 'button', focus zone2 unmounts. And inspecting zone1 shows that the wrapper element in zone1 still has aria-hidden=true.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/18fea2368512d81c9ed502fdac42e668069ef30d'>18fea2368512d81c9ed502fdac42e668069ef30d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by ling1726** [comment]
It seems this could reproduced even more simply https://codesandbox.io/s/intelligent-wescoff-610heq?file=/example.tsx

In the above example I open the dialog from the popup, Then after I close the dialog, the entire screen still seems to be `aria-hidden`

image

**Review by ling1726 — APPROVED** [review]
